Library/Homebrew/formula.rb

```diff
@@ -1745,14 +1745,14 @@ def generate_completions_from_executable(*commands,
       elsif shell_parameter_format == :arg
         "--shell=#{shell}"
       elsif shell_parameter_format == :none
-        ""
+        nil
       else
         "#{shell_parameter_format}#{shell}"
       end
 
       popen_read_args = %w[]
       popen_read_args << commands
-      popen_read_args << shell_parameter
+      popen_read_args << shell_parameter unless shell_parameter.nil?
```

Comment:
```suggestion
      popen_read_args << shell_parameter unless shell_parameter.blank?
```

`"".blank?` is true so this will take core of more edge cases (e.g. `shell_parameter_format: ""`)

Reply:
> `"".blank?` is true so this will take core of more edge cases (e.g. `shell_parameter_format: ""`)

However, `shell_parameter_format: ""` will jump to line 1750, resulting in `shell_parameter = "#{shell}"`.

Nevertheless, I agree it reads more clearly.